src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int index = getIndex(key);
+        Node<K, V> node = table[index];
 
+        if (node == null) {
+            table[index] = new Node<>(key, value);
+            size++;
+        } else {
+            while (true) {
+                if (Objects.equals(key, node.key)) {
+                    node.value = value;
+                    return;
+                }
+                if (node.next == null) {
+                    node.next = new Node<>(key, value);
+                    size++;
+                    return;
+                }
+                node = node.next;
+            }
+        }
```

Comment:
```suggestion
        Node<K, V> node = table[index];
        while (node != null) {
             if (Objects.equals(key, node.key)) {
                    node.value = value;
                    return;
             }
             node = node.next;
        }
        table[index] = new Node<>(key, value, table[index]);
        size++;
```
while (true) is not a good practice
we can rewrite this method like this